frontend/src/components/VideoCarrouselSlick.js

```diff
@@ -35,24 +36,50 @@ class VideoCarrouselSlick extends Component {
         }
     }
 
-    afterChangeMethod() {
-        const nextCarrousselCount = this.state.carrousselCount + 1;
-        const pageCount = nextCarrousselCount * this.CARROUSSEL_SIZE / this.VIDEO_PER_PAGE;
-        if (pageCount === this.state.apiCallCount && pageCount <= this.state.pagesTotal) {
-            const nextApiCount = this.state.apiCallCount + 1;
-            djangoAPI.get(`/videos?page=${nextApiCount}`)
-                .then((response) => {
-                    let video = this.state.videos;
-                    video.push(...response.data.results);
-                    this.setState({
-                        videos: video,
-                        apiCallCount: nextApiCount,
-                        carrousselCount: nextCarrousselCount
-                    });
-                });
+    /**
+     * Select the call to make to the api depending if the search bar is used or not
+     * @param queryText
+     * @param nextApiCount
+     * @returns {Promise<AxiosResponse<any>>} response from the API call
+     */
+    async setApICall(queryText, nextApiCount){
+        let response;
+        if (queryText !== '') {
+            response = await djangoAPI.get(`/search_video/?page=${nextApiCount}`, {
+                params: {
+                    q: queryText
+                }
+            });
         }
         else {
-            this.setState({ carrousselCount: nextCarrousselCount });
+            response = await djangoAPI.get(`/videos?page=${nextApiCount}`);
+        }
+        return response;
+    }
+
+    /**
+     * this method is called by react slick after the slider finish transition
+     * used to compute if we need to make new API calls
+     * @param index
+     * @returns {Promise<void>}
+     */
+    async afterChangeMethod(index) {
+        //index is gave by react slick and correspond to the index of the video on the left (start at 1)
+        const nextCarrousselCount = index > this.state.index ? this.state.carrousselCount + 1 : this.state.carrousselCount - 1;
+        //we add 5 to index to calcultate the number of videos displayed so far
+        const pageCount = (index + 5) / this.VIDEO_PER_PAGE;
+        if(pageCount === this.state.apiCallCount && pageCount <= this.state.pagesTotal){
```

Comment:
We could use the number of video in the previous videos request response for calculating pageCount.
Also, we should check if we are at the end of the paginator in a different way than using `pageCount <= this.state.pagesTotal`.
I think we can, after recieving a videos response, check if we are at the end of the paginator.
The back gives num_pages. So we should just compare it to the page we requested.
If they are equal we set a flag so we know we are at the end from the front.
If I understand the variables correctly, something like (not sure if apiCallCount refers to the number of paginator page -backend- we query):
`let lastPageLoaded = ((this.apiCallCount === response.data.num_pages) ? true : false);`
Then we can put that in the state, and probably deactivate the button to slide the carousel.